### DISCUSSION TITLE
TEKSI Module Management Tool (TMMT)

### DISCUSSION THREAD

**Discussion by ponceta** [teksi/TMMT#1] [Ideas]
**Context**
- In 2016, we decided to keep track of all datamodel alterations on the QWAT projet (drinking water module)
- In 2018, we introduced this necessity to the QGEP project (wastewater module)
- To avoid script duplication, the PUM tool ([Postgres Update Manager](https://github.com/opengisch/pum)) is created as a common command line tool
- In 2020 Windows integration of PUM is included to support microsoft users (still in command line)
- In 2021 a prototype of a datamodel Tool is created on the QGEP project to ease the visual management of updates
- In 2023 this prototype is documented and is considered as a very easy approach for providers (https://qgep.github.io/docs/fr/admin-guide/datamodel-tool/index.html)

**Knowing this**

We want to provide users a generic management tool that would be able :
- [x] to initialize and help the usage roles and users assignement
- [x] to intialize any TEKSI module datamodel https://github.com/teksi/TMMT/issues/3
- [x] to check any TEKSI module datamodel version https://github.com/teksi/TMMT/issues/4
- [x] to upgrade any TEKSI module datamodel https://github.com/teksi/TMMT/issues/5
- [x] to provide .qgs project to any TEKSI module

On the idea of the actual QGIS plugin https://github.com/teksi/TMMT/issues/2 allowing the user 

1. To choose the module he wants to install/check/upgrade
2. To choose the database he wants to use
3. To choose the specific version he wants to upgrade to

**Specifications**
To be technically defined but in the idea :
An independant QGIS plugin, downloadable on the QGIS plugin library, Community driven (template is provided any new module is welcome)

**Easy to configure**
- [x] It should be documented on how to add some new release, [new module to the Tool](https://github.com/opengisch/oqtopus/blob/main/docs/docs/module_development/adding_modules.md)

![image](https://github.com/teksi/Home/assets/8614223/0bb75667-155e-42cb-8575-83449d143662)

**Comment by w0pr** [comment]
How does the oqtopus plugin relate to TMMT and where does it fit into the timeline? As I see it, it provides about the same functionality as TMMT and is based on it (as mentioned in the releases). I currently use oqtopus to deploy TEKSI wastewater; can or should I switch over to TMMT?

**Reply by ponceta** [reply]
TMMT is a TEKSI branding of oQtopus which is the TEKSI developed tool to deploy and QGIS/postgres/postgis datamodel. You should be able to do it with one or another without limitation. The only actual limitation is technical, QGIS doesn't like both plugins installed, just choose the one you prefer.

**Reply by ponceta** [reply]
The main idea is to make possible to search for 'TEKSI' in the QGIS plugin repository. Right now both are in experimental versions.
